### PR TITLE
Add migration guide for buildpack api 0.6

### PR DESCRIPTION
**Pull request by natalieparellano**
Maybe blocked on lifecycle 0.11.0

Signed-off-by: Natalie Arellano <narellano@vmware.com>

### CODE REVIEW

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
++++
+title="Buildpack API 0.5 -> 0.6"
++++
+
+<!--more-->
+
+This guide is most relevant to buildpack authors.
+
+See the [spec release](https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.6) for buildpack API 0.6 for the full list of changes and further details.
```
Comment (yaelharel) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:9]: Need to release it only after the spec is released (the link is currently broken).

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+### Opt-in layer caching
+
+Buildpacks must now explicitly opt-in to layer re-use by writing `launch`, `build`, or `cache` keys to a new `[types]` table in `<layers>/<layer>.toml` (note that these keys were removed from the top level). If buildpacks do not modify `<layers>/<layer>.toml`, the layer will not be re-used, even if the buildpack in the previous build set any of these keys to `true`.
+
+A Bash buildpack could write something like the following to `<layers>/<layer>.toml` in order to cache a layer:
+
+```bash
+cat >> layer.toml <<EOL
+[types]
+cache = true
```
Comment (yaelharel) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:20]:
```suggestion
  cache = true
```
I think that the 2 spaces are necessary but I'm not 100% sure.
Anyway, I think it's clearer to have these spaces.

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+EOL
+```
+
+### Buildpacks contribute default process type
+
+Buildpacks may now contribute the default process type by writing `default = true` in the process type definition in `<layers>/launch.toml`. An individual buildpack may only specify one process type with `default = true`. The lifecycle will choose, from all buildpack-provided process types, the last process type with `default = true` as the buildpack-provided default. A user may override the buildpack-provided default process type by passing `-process-type` to the exporter. (Note: to enable buildpacks implementing older Buildpack APIs to work with newer buildpacks, the lifecycle will assume that buildpacks on Buildpack API < 0.6 intended for `web` processes to be the default.)
```
Comment (yaelharel) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:26]:
```suggestion
Buildpacks may now contribute the default process type by writing `default = true` in the process type definition in `<layers>/launch.toml`. An individual buildpack may only specify one process type with `default = true`. The lifecycle will choose, from all buildpack-provided process types, the last process type with `default = true` as the buildpack-provided default. A user may override the buildpack-provided default process type by passing `-process-type` to the exporter. Note: to enable buildpacks implementing older Buildpack APIs to work with newer buildpacks, the lifecycle will assume that buildpacks on Buildpack API < 0.6 intended for `web` processes to be the default.
```

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+### exec.d on Windows
+Buildpack API 0.5 introduced support for `exec.d` executables that may be used to modify the environment for an app process (similar to profile.d) but do not depend on a shell.
+The launcher will execute each file in each `<layers>/<layer>/exec.d` directory and each file in each `<layers>/<layer>/exec.d/<process>` directory before the app process is started.
+Previously, this was only implemented on Linux, but it is now implemented on Windows as well. On Windows, the launcher will pass each `exec.d` executable, via the `CNB_EXEC_D_HANDLE` environment variable, the hex representation of an open file handle where the executable is expected to write its output. The output may contain any number of top-level key/value pairs, which will be sourced by the launcher in the app process's environment.
```
Comment (yaelharel) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:39]:
```suggestion
Previously, this was only implemented on Linux, but starting from Buildpack API 0.6, it is implemented on Windows as well. On Windows, the launcher will pass each `exec.d` executable, via the `CNB_EXEC_D_HANDLE` environment variable, the hex representation of an open file handle where the executable is expected to write its output. The output may contain any number of top-level key-value pairs, which will be sourced by the launcher in the app process's environment.
```

I think that the second sentence (starting with "On Windows,") in this paragraph is too long. Can we break it to two sentences?

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md (hunk 2 above)
Comment (natalieparellano) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:20]: After discussion we think the spaces are not necessary.

**Review by yaelharel — APPROVED** [review]

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+title="Buildpack API 0.5 -> 0.6"
++++
+
+<!--more-->
+
+This guide is most relevant to buildpack authors.
+
+See the [spec release](https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.6) for buildpack API 0.6 for the full list of changes and further details.
+
+### Opt-in layer caching
```
Comment (samj1912) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:11]: Should this and the fact that the layer types have been moved under a types table be two separate entries?

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+```bash
+cat >> layer.toml <<EOL
+[types]
+cache = true
+EOL
+```
+
+### Buildpacks contribute default process type
+
+Buildpacks may now contribute the default process type by writing `default = true` in the process type definition in `<layers>/launch.toml`. An individual buildpack may only specify one process type with `default = true`. The lifecycle will choose, from all buildpack-provided process types, the last process type with `default = true` as the buildpack-provided default. A user may override the buildpack-provided default process type by passing `-process-type` to the exporter. Note: to enable buildpacks implementing older Buildpack APIs to work with newer buildpacks, the lifecycle will assume that buildpacks on Buildpack API < 0.6 intended for `web` processes to be the default.
```
Comment (samj1912) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:26]: Should we add an example here?

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
```
Comment (samj1912) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:27]:
```suggestion
For example - 

```toml
# launch.toml
[[processes]]
type = "<process type>"
command = "<command>"
args = ["<arguments>"]
direct = false
# the default field is new
default = false
```
```

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md (hunk 5 above)
Comment (yaelharel) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:11]: I agree. It's also 2 separate entries in the spec: https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.6.

Comment (natalieparellano) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:11]: I don't know... the fact that it's two separate issues is an artifact of how the RFC unfolded (see comment: https://github.com/buildpacks/spec/pull/151#issuecomment-708631142).

Comment (natalieparellano) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:11]: However I do think the paragraph below could be re-worded...taking a look.

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+```toml
+# launch.toml
+[[processes]]
+type = "<process type>"
+command = "<command>"
+args = ["<arguments>"]
+direct = false
+# the default field is new
+default = false
```
Comment (natalieparellano) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:37]:
```suggestion
default = false
```
```

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md (hunk 6 above)
Comment (natalieparellano) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:26]: Thanks for the suggestion!

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md (hunk 5 above)
Comment (natalieparellano) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:11]: How does it look now?

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+title="Buildpack API 0.5 -> 0.6"
++++
+
+<!--more-->
+
+This guide is most relevant to buildpack authors.
+
+See the [spec release](https://github.com/buildpacks/spec/releases/tag/buildpack%2Fv0.6) for buildpack API 0.6 for the full list of changes and further details.
+
+### Opt-in layer caching
```
Comment (samj1912) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:11]:
```suggestion
### Opt-in layer caching and launch.toml format changes
```

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+Buildpacks must now explicitly opt-in to layer re-use by setting the `launch`, `build`, and `cache` keys in `<layers>/<layer>.toml`. If buildpacks do not modify `<layers>/<layer>.toml`, the layer will behave like a temporary directory, available only to the authoring buildpack, existing for the duration of a single build - even if the buildpack in the previous build set any of these keys to `true`.
+
+Additionally, the `launch`, `build`, and `cache` keys are moved under a new `[types]` table in `<layers>/<layer>.toml` and removed from the top level. A Bash buildpack could write something like the following to `<layers>/<layer>.toml` in order to cache a layer:
```
Comment (samj1912) [inline content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:15]:
```suggestion
> **NOTE**: Additionally, the `launch`, `build`, and `cache` keys are moved under a new `[types]` table in `<layers>/<layer>.toml` and removed from the top level. 

A Bash buildpack could write something like the following to `<layers>/<layer>.toml` in order to cache a layer:
```


Does this work better? 🤔

**Review by samj1912 — APPROVED** [review]

File: content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md (hunk 10 above)
Comment (natalieparellano) [reply content/docs/reference/spec/migration/buildpack-api-0.5-0.6.md:15]: I like it!

**Review by samj1912 — APPROVED** [review]

**Review by jromero — APPROVED** [review]

**Pull request merged**